### PR TITLE
Implement `DataFrame.first` and `Series.first` functionality

### PR DESCRIPTION
**Pull request by awdavidson**
Please see change to implement DataFrame.first and Series.first functionality similar to that available in pandas. Requirement raised in issue: #1929

```python
>>> index = pd.date_range('2018-04-09', periods=4, freq='2D')
>>> ks_series = ks.Series([1, 2, 3, 4], index=index)
2018-04-09  1
2018-04-11  2
2018-04-13  3
2018-04-15  4
dtype: int64

>>> ks_series.first('3D')
2018-04-09  1
2018-04-11  2
dtype: int64
```

### CODE REVIEW

File: databricks/koalas/frame.py [hunk 1]
```diff
@@ -3077,14 +3077,12 @@ def between_time(
...
         """
-        from databricks.koalas.indexes import DatetimeIndex
-
         axis = validate_axis(axis)
 
         if axis != 0:
             raise NotImplementedError("between_time currently only works for axis=0")
 
-        if not isinstance(self.index, DatetimeIndex):
+        if not isinstance(self.index, ks.DatetimeIndex):
```
Comment (ueshin) [inline databricks/koalas/frame.py:3085]: Could you also update it in `at_time`?

File: databricks/koalas/tests/test_series.py [hunk 2]
```diff
@@ -188,6 +188,14 @@ def test_last(self):
             self.kser.last("1D")
         self.assert_eq(ks_input.last("1D"), pd_input.last("1D"))
 
+    def test_first(self):
+        index = pd.date_range("2018-04-09", periods=4, freq="2D")
+        pd_input = pd.Series([1, 2, 3, 4], index=index)
+        ks_input = ks.Series([1, 2, 3, 4], index=index)
```
Comment (ueshin) [inline databricks/koalas/tests/test_series.py:194]: Shall we use `ks.from_pandas`?
Could you also update it in `test_last`?

Comment (ueshin) [reply databricks/koalas/tests/test_series.py:194]: Btw, we usually use `pser` or `kser` for `Series` in tests. Could you follow it?

**Review by ueshin — COMMENTED** [review]
@awdavidson Thanks for working on this!
I left some comments. Could you address them? Thanks!

File: databricks/koalas/tests/test_series.py (hunk 2 above)
Comment (awdavidson) [reply databricks/koalas/tests/test_series.py]: Sure will make the change :) Apologies for the inconsistency!

File: databricks/koalas/frame.py (hunk 1 above)
Comment (awdavidson) [reply databricks/koalas/frame.py:3085]: > @awdavidson Thanks for working on this!
> I left some comments. Could you address them? Thanks!

No problem. Comments have been address :)

**Comment by awdavidson** [comment]
> @awdavidson Thanks for working on this!
> I left some comments. Could you address them? Thanks!

No problem, comments have been addressed. Thanks! :)

File: databricks/koalas/tests/test_series.py (hunk 2 above)
Comment (ueshin) [reply databricks/koalas/tests/test_series.py]: No worries! Thanks for the change!

**Review by ueshin — APPROVED** [review]
LGTM, pending tests.

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**